static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -0,0 +1,301 @@
+        }
+      : undefined
+  );
+
+  const {mutate: updateViews} = useUpdateGroupSearchViews();
+
+  const debounceUpdateViews = debounce((newTabs: Tab[]) => {
+    if (newTabs) {
+      updateViews({
+        orgSlug: organization.slug,
+        groupSearchViews: newTabs.map(tab => ({
+          // Do not send over an ID if it's a temporary id
+          ...(tab.id[0] !== '_' ? {id: tab.id} : {}),
+          name: tab.label,
+          query: tab.query,
+          querySort: tab.querySort,
+        })),
+      });
+    }
+  }, 500);
+
+  useEffect(() => {
+    // If no query, sort, or viewId is present, set the first tab as the selected tab, update query accordingly
+    if (!query && !sort && !viewId) {
+      navigate({
+        query: {
+          ...queryParams,
+          query: draggableTabs[0].query,
+          sort: draggableTabs[0].querySort,
+          viewId: draggableTabs[0].id,
+        },
+        pathname: `/organizations/${organization.slug}/issues/`,
+      });
+      return;
+    }
+    // if a viewId is present, check if it exists in the existing views.
+    if (viewId) {
+      const selectedTab = draggableTabs.find(tab => tab.id === viewId);
+      if (
+        selectedTab &&
+        (query !== selectedTab!.query || sort !== selectedTab!.querySort)
+      ) {
+        // if a viewId exists but the query and sort are not what we expected, set them as unsaved changes
+        setDraggableTabs(
+          draggableTabs.map(tab =>
+            tab.key === selectedTab!.key
+              ? {
+                  ...tab,
+                  unsavedChanges: [query, sort],
+                }
+              : tab
+          )
+        );
+      } else if (!selectedTab) {
+        // if a viewId does not existing, remove it from the query
+        navigate({
+          query: {
+            ...queryParams,
+            viewId: undefined,
+          },
+          pathname: `/organizations/${organization.slug}/issues/`,
+        });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Update local tabs when new views are received from mutation request
+  useEffect(() => {
+    setDraggableTabs(
+      draggableTabs.map(tab => {
+        if (tab.id[0] === '_') {
+          // Temp viewIds are prefixed with '_'
+          views.forEach(view => {
+            if (
+              view.id &&
+              tab.query === view.query &&
+              tab.querySort === view.querySort &&
+              tab.label === view.name
+            ) {
+              tab.id = view.id;
+            }
+          });
+          navigate({
+            query: {
+              ...queryParams,
+              viewId: tab.id,
+            },
+            pathname: `/organizations/${organization.slug}/issues/`,
+          });
+        }
+        return tab;
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [views]);
+
+  // Updates the tab's hasSavedChanges state
```

Comment:
Actually no, it's a little more complicated since we are maintaining unsaved changes when you switch tabs 🤔 

I think this is probably the right call then